src/circuit/circuit_frame.rs

```diff
@@ -1463,12 +1559,20 @@ fn reduce_cons<F: PrimeField, CS: ConstraintSystem<F>>(
 
     results.add_clauses_cons(*if_hash.value(), &arg1, env, &continuation, &g.false_num);
 
-    {
-        // head == CURRENT-ENV
-        // FIXME: Error if rest != NIL.
-
-        results.add_clauses_cons(*current_env_hash.value(), env, env, cont, &g.true_num);
-    }
+    // head == CURRENT-ENV
+    let the_cont_if_rest_is_nil = AllocatedContPtr::pick(
+        &mut cs.namespace(|| "the_cont_if_rest_is_nil"),
+        &rest_is_nil,
+        cont,
+        &g.error_ptr_cont,
+    )?;
```

Comment:
This makes me realize (and I just checked) that we don't actually correctly handle cases where no body is supplied to `let` and `letrec`.

For example,

`(let)` and `(let ((a 1)))` 'should' both be errors, but in fact they return NIL. By 'should' I mean that is what `lurk/api.lisp` defines.

FWIW, Common Lisp makes `(let)` an error but `(let ())` is NIL. The latter is arguably fine.

If we were to define NIL as the default value in such situations, we should probably do the same for IF with only branch. @namin Do you have an opinion on this?

Reply:
We're being explicit about `begin`s as well, so the body of the `let`/`letrec` or even `lambda` should be exactly one expression. The spec is clear on that.

In Scheme, `(let ())` is invalid syntax.

I don't see much benefit of relaxing the spec. Except perhaps for explicit `begin`, but we don't have all that many uses for it.

Reply:
This makes sense to me. @emmorais Can you add this to the current PR?
- tests for `(let)`, `(let ((a 1)))`, `(letrec)`, and `(letrec ((a 1)))` — all of which should be an error.
- implement in `eval.rs`
- implement circuit (with tests)

That's quite a few things, but it should be very mechanical and pretty easy at this point.

Then I think all the error conditions we know about will be accounted for :) .